dask/dataframe/utils.py

```diff
@@ -201,9 +201,9 @@ def has_known_categories(x):
     x : Series or CategoricalIndex
     """
     x = getattr(x, "_meta", x)
-    if isinstance(x, pd.Series):
+    if is_series_like(x):
         return UNKNOWN_CATEGORIES not in x.cat.categories
-    elif isinstance(x, pd.CategoricalIndex):
+    elif is_index_like(x) and hasattr(x, "cat"):
```

Comment:
I think hasattr(x, "cat") isn't quite the same logic, as pd.CategoricalIndex doesn't have that attribute. So passing a categorical index will lead to `elif True and False` which evaluates to False, rather than True as it did before

Reply:
You are correct, I deviated from preserving the logical correctness here. cudf CategoricalIndex too has `categories` after having a discussion offline changed it to `categories`.